data/transactions/logic/eval.go

```diff
@@ -501,7 +489,7 @@ type EvalContext struct {
 	programHashCached crypto.Digest
 
 	// Stores state & disassembly for the optional debugger
-	debugState DebugState
+	debugState *DebugState
```

Comment:
@jannotti For my understanding:  How did you determine it makes sense to `debugState` from struct field value to pointer?

I'm asking because the value vs pointer distinction isn't something I've had to think about in some time.  So, I'm trying to improve my heuristics for appropriate usage.

From what I can tell in the PR, it looks like `debugState` usage relies on modifying the `EvalContext` instance.  So, I'm _guessing_ the change makes the intended usage pattern clearer to the reader.

Reply:
@jannotti Thanks for the explanation in https://github.com/algorand/go-algorand/pull/3612#issuecomment-1040390583.  Makes sense - Resolving.